packages/terminal-extension/src/index.ts

```diff
@@ -204,31 +208,31 @@ export function addCommands(
     label: args => (args['isPalette'] ? 'New Terminal' : 'Terminal'),
     caption: 'Start a new terminal session',
     iconClass: args => (args['isPalette'] ? '' : TERMINAL_ICON_CLASS),
-    execute: args => {
+    execute: async args => {
+      // wait for the widget to lazy load
+      const { Terminal } = await Private.ensureWidget();
       const name = args['name'] as string;
       const term = new Terminal();
-      const promise = name
-        ? serviceManager.terminals
-            .connectTo(name)
-            .catch(() => serviceManager.terminals.startNew())
-        : serviceManager.terminals.startNew();
 
       term.title.icon = TERMINAL_ICON_CLASS;
       term.title.label = '...';
       let main = new MainAreaWidget({ content: term });
       app.shell.add(main);
 
-      return promise
-        .then(session => {
-          term.session = session;
-          tracker.add(main);
-          app.shell.activateById(main.id);
-
-          return main;
-        })
-        .catch(() => {
-          term.dispose();
-        });
+      try {
+        term.session = await (name
```

Comment:
this is actually a little easier to track, though...

Reply:
Yeah, I think this is clearer.